guacamole/src/main/webapp/app/form/controllers/languageFieldController.js

```diff
@@ -40,7 +40,7 @@ angular.module('form').controller('languageFieldController', ['$scope', '$inject
 
     // Retrieve defined languages
     languageService.getLanguages().then(function languagesRetrieved(languages) {
-        $scope.$apply(function updateLanguageOptions() {
+        $scope.$applyAsync(function updateLanguageOptions() {
```

Comment:
I'm not really sure this change should be, here.  We can talk about changing this; however, I believe it is out of scope for a set of changes targeted at improving Chinese language support.

Reply:
This is a bug that I fixed that would cause an existing script error, I fixed it, and if I don't modify it here, my commit may not work.

I did a simple test, it was fixed, and it worked fine.

Unfortunately I don't have the ability to write unit tests for this modification at this time because it is beyond my ability to do so.

Reply:
I have explained the reason for this change in jira.

I found that there is a script error in the front end when setting the default internationalization. This error is caused by an ajax language available in angular. I checked the information and fixed the problem.

Reply:
> I have explained the reason for this change in JIRA.

Not yet - the JIRA issue currently says "fixed ... replace "$apply" to "$applyAsync"" and links to this commit. Please clarify on the issue, when you can, the actual underlying nature of the issue being fixed.

> I found that there is a script error in the front end when setting the default internationalization. This error is caused by the wrong use of angular ajax request.

`$apply()` and `$applyAsync()` do not deal with AJAX, nor is AJAX involved with the internationalization system. The error in question (infinite digest) results from `$apply()` being used within a digest cycle.

I see the same errors you do, but please describe on the issue the nature of the issue being fixed (why current usage of `$apply()` leads to an infinite digest error). The reason that `$apply()` is incorrect here (ie: what circumstances would cause this particular call to be invoked within an active digest) needs to be known.

Splitting this specific fix into its own pull request (rather than piggybacking it on GUACAMOLE-903) should also be done.

> Unfortunately I don't have the ability to write unit tests for this modification at this time because it is beyond my ability to do so.

I'm not sure what unit test would need to be written for this, but don't worry - there's no need in this case. All that is needed is (1) to split this change such that it only covers the scope of the specific issue and (2) to document the nature of the issue in JIRA.

Reply:
okay!I'll do as you say. I will open another PR after this PR merge to submit this bug fix.